app/upgrades/v17/upgrades.go

```diff
@@ -133,3 +150,48 @@ func CreateUpgradeHandler(
 		return migrations, nil
 	}
 }
+
+// FlipTwapSpotPriceRecords flips the denoms and spot price of twap record of a given pool.
+func FlipTwapSpotPriceRecords(ctx sdk.Context, poolIds []uint64, keepers *keepers.AppKeepers) error {
+	for _, poolId := range poolIds {
+		// check that this is a cl pool
+		_, err := keepers.ConcentratedLiquidityKeeper.GetConcentratedPoolById(ctx, poolId)
```

Comment:
Lets just take in an array of PoolI without the type check. Then in the upgrade handler we just call GetPools and pass in the result of GetPools directly in. No need to
1. Get pools
2. Extract pool IDs
3. Get pool again

Reply:
fixed! https://github.com/osmosis-labs/osmosis/pull/5939/commits/21a755262a9715599dae1175a6ad3a11f55f1bb3